tuf/utils/role_sort.go

```diff
@@ -0,0 +1,28 @@
+package utils
+
+import (
+	"strings"
+)
+
+// RoleList is a list of roles
+type RoleList []string
+
+// Len returns the length of the list
+func (r RoleList) Len() int {
+	return len(r)
+}
+
+// Less returns true if the item at i should be sorted
```

Comment:
It doesn't hurt if we make this also alphabetically ordered within the same length bucket. The potential to have non-reproducible bugs because the final ordering is non-deterministic should overweight whatever CPU penalty we take for the alphabetic ordering operations.